### PR TITLE
SNOW-2467884: Add parameter binding feature tests

### PR DESCRIPTION
**Pull request by sfc-gh-rkowalski**
Created a new `ParameterBindingTests.java` file containing comprehensive JDBC test cases for parameter binding functionality. The tests cover basic type binding with positional parameters, table operations (insert, update, delete, select), edge cases with NULL values and special characters, and complex scenarios with multiple parameters.

### CODE REVIEW

**Comment by sfc-gh-rkowalski** [comment]
* **#459** <a href="https://app.graphite.com/github/pr/snowflakedb/universal-driver/459?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/snowflakedb/universal-driver/459?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#458** <a href="https://app.graphite.com/github/pr/snowflakedb/universal-driver/458?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#457** <a href="https://app.graphite.com/github/pr/snowflakedb/universal-driver/457?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Review by sfc-gh-merbel — APPROVED** [review]

File: jdbc/src/test/java/net/snowflake/jdbc/e2e/query/ParameterBindingTests.java [hunk 1]
```diff
@@ -0,0 +1,491 @@
...
+public class ParameterBindingTests extends SnowflakeIntegrationTestBase {
+  @Test
+  public void shouldBindBasicTypesWithPositionalParameters() throws Exception {
+    // Given Snowflake client is logged in
+    Connection connection = getDefaultConnection();
+
+    // When Query "SELECT ?, ?, ?, ?, ?" is executed with positional parameters
+    // [42, 3.14, "hello", True, None]
+    String sql = "SELECT ?, ?, ?, ?, ?";
+    try (PreparedStatement preparedStatement = connection.prepareStatement(sql)) {
```
Comment (sfc-gh-dheyman) [inline jdbc/src/test/java/net/snowflake/jdbc/e2e/query/ParameterBindingTests.java:30]: You can use `withPreparedQueryResult` from the base class.

File: jdbc/src/test/java/net/snowflake/jdbc/e2e/query/ParameterBindingTests.java [hunk 2]
```diff
@@ -0,0 +1,491 @@
...
+      preparedStatement.setInt(1, 28);
+      try (ResultSet resultSet = preparedStatement.executeQuery()) {
+        // Then Result should contain rows for "Alice" and "Charlie"
+        List<String> names = new ArrayList<>();
+        while (resultSet.next()) {
+          names.add(resultSet.getString(2));
+        }
+        assertEquals(2, names.size(), "Expected two matching rows");
+        assertTrue(names.contains("Alice"), "Expected Alice in filtered results");
```
Comment (sfc-gh-dheyman) [inline jdbc/src/test/java/net/snowflake/jdbc/e2e/query/ParameterBindingTests.java:257]: if you add `ORDER BY` to the SQL, you won't have to do the `contains` lookups.

File: jdbc/src/test/java/net/snowflake/jdbc/e2e/query/ParameterBindingTests.java [hunk 3]
```diff
@@ -0,0 +1,491 @@
...
+      preparedStatement.setString(2, "Charlie");
+      preparedStatement.executeUpdate();
+    }
+
+    // And Query "SELECT * FROM table ORDER BY id" is executed
+    String selectSql = "SELECT * FROM " + tableName + " ORDER BY id";
+    try (Statement statement = connection.createStatement();
+        ResultSet resultSet = statement.executeQuery(selectSql)) {
+      // Then Result should contain 3 rows with correct values
+      assertTrue(resultSet.next(), "Expected first row");
```
Comment (sfc-gh-dheyman) [inline jdbc/src/test/java/net/snowflake/jdbc/e2e/query/ParameterBindingTests.java:146]: in other tests I can see `wasNull()`. I suggest being consistent.
